tensilelite/Tensile/KernelWriter.py

```diff
@@ -2604,28 +2617,19 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
       if (kernel["AssertSummationElementMultiple"] % KinInnerUnroll == 0):
         tailLoopInnerUnroll = kernel["InnerUnroll"]
 
-      # need to unroll tail loop for the following cases
-      mEnd = 1
-      if kernel["ProblemType"]["Sparse"]:
-        mEnd = kernel["LoopIters"]
-      if (kernel["DirectToVgprA"] or kernel["DirectToVgprB"]):
-        mEnd = kernel["DepthU"]//(kernel["MatrixInstK"]*kernel["LocalSplitU"])
-      elif kernel["DirectToLds"] and kernel["EnableMatrixInstruction"] and kernel["InnerUnroll"] == 1 and\
-            (kernel["GlobalReadVectorWidthA"] * self.states.bpeAB > 4 or kernel["GlobalReadVectorWidthB"] * self.states.bpeAB > 4) and \
-            kernel["DepthU"] // kernel["MatrixInstK"] > 2:
-        mEnd = kernel["DepthU"] // (kernel["MatrixInstK"] * 2)
-
-      # remove vgprBuffer for local read from vgprPool because we are ready to issue local read
-      self.vgprPool.remove(self.states.a.startVgprValu , \
-        self.states.lastValuAB - self.states.a.startVgprValu , "ValuAB") # remove from pool
-      module.addComment1("Tail: remove ValuA/B vgpr buffer [%u...%u) from pool" % \
-                        (self.states.a.startVgprValu , self.states.lastValuAB))
+      # TailLoop unroll case, we need to keep these vgpr
+      if mEnd == 1:
+        # remove vgprBuffer for local read from vgprPool because we are ready to issue local read
+        self.vgprPool.remove(self.states.a.startVgprValu , \
+          self.states.lastValuAB - self.states.a.startVgprValu , "ValuAB") # remove from pool
+        module.addComment1("Tail: remove ValuA/B vgpr buffer [%u...%u) from pool" % \
+                          (self.states.a.startVgprValu , self.states.lastValuAB))
 
       # add address vgpr to vgprPool
-      self.vgprPool.add(self.states.lastValuAB , \
-        self.states.lastVgprForReads - self.states.lastValuAB, "address vgpr") # Add as available
-      module.addComment1("Tail: add address/G2L vgpr [%u...%u) to pool" % \
-                        (self.states.lastValuAB, self.states.lastVgprForReads))
+        self.vgprPool.add(self.states.lastValuAB , \
+          self.states.lastVgprForReads - self.states.lastValuAB, "address vgpr") # Add as available
+        module.addComment1("Tail: add address/G2L vgpr [%u...%u) to pool" % \
+                          (self.states.lastValuAB, self.states.lastVgprForReads))
```

Comment:
Should probably indent the comment as well to match, indentation here has lots of variance making it more difficult to see what's inside the block